onnxruntime/core/graph/function.cc

```diff
@@ -7,20 +7,15 @@
 #include "onnx/shape_inference/implementation.h"
 
 namespace onnxruntime {
-void TypeConstraintHelper(const ONNX_NAMESPACE::FunctionProto* onnx_func_proto_,
+void IOTypeConstraintHelper(const ONNX_NAMESPACE::FunctionProto* onnx_func_proto_,
```

Comment:
I don't understand the purpose of the part of this function that infers type-constraints for the inputs/outputs of the function here. I realize that this is pre-existing code and not part of the current PR. But I see various issues and concerns with that old functionality here. Where does this eventually get used? I think we might be able to fix that piece of code and avoid having to generate these type-constraints here. Or, at least we should document this as a TODO and come back clean this up eventually.

Reply:
We registered a type constraint for add input/output to the generated tensor. Tried remove the type constraint generation and the test case failed.. for now lets keep it there. I added a to-do here. 
A more relaxing way is to add all type as allowed type.